### PR TITLE
B#1111 User assets sometimes don't load

### PR DESCRIPTION
**Pull request by gdamron**
- Check if an asset is already in the manifest before adding to avoid an exception
- Allow '-' characters in schema queries in order to search for guids
- Signal that elements using an asset should reload after adding an asset since there is no guarantee before that the asset was added before the scene was updated. If it was already updated, the asset could not have loaded.

### CODE REVIEW

File: Assets/Source/Player/Asset/AssetManifest.cs [hunk 1]
```diff
@@ -209,6 +209,16 @@ public void Destroy()
             _guidToRecord.Clear();
         }
 
+        /// <summary>
+        /// Returns true if an asset exists in the manifest.
+        /// </summary>
+        /// <param name="asset"></param>
+        /// <returns></returns>
+        public bool Contains(AssetData asset)
```
Comment (thegoldenmule) [inline Assets/Source/Player/Asset/AssetManifest.cs]: Please add appropriate unit test coverage for this, as `AssetManifest` is a central system.

File: Assets/Source/Player/IUX/Element/ElementQuery.cs [hunk 2]
```diff
@@ -22,7 +22,7 @@ internal class ElementQuery
         /// <summary>
         /// Regex for property matches.
         /// </summary>
-        private static readonly Regex PROPERTY_QUERY = new Regex(@"^\(@([\w-_\.]+)([<>=!]=?)(\w+)\)$");
+        private static readonly Regex PROPERTY_QUERY = new Regex(@"^\(@([\w-_\.]+)([<>=!]=?)([\w-]+)\)$");
```
Comment (thegoldenmule) [inline Assets/Source/Player/IUX/Element/ElementQuery.cs]: Add a unit test that tests guid lookup.

File: Assets/Source/Player/Asset/AssetManifest.cs (hunk 1 above)
Comment (gdamron) [reply Assets/Source/Player/Asset/AssetManifest.cs]: 👍

File: Assets/Source/Player/IUX/Element/ElementQuery.cs (hunk 2 above)
Comment (gdamron) [reply Assets/Source/Player/IUX/Element/ElementQuery.cs]: 👍

**Comment by gdamron** [comment]
yes, all unit tests still pass with the changes. it's worth pointing out that our current tests use values like "guid" or single letters for guids, which is why they never illustrated that searching for an _actual_ guid fails.

**Comment by gdamron** [comment]
tests updated

**Pull request merged**